Connect to each channel in PublicResponses automatically.

murnux opened the pull request:
So this adds the functionality desired from issue #14. 

I am only including PublicResponses in this pull for 2 reasons:
- See if others think this is a reasonable solution.
- I wasn't sure if there would be a realistic possibility of there being channels in PublicResponses that wouldn't be in dmRepsonses and ephResponses. If this can occur, I'll work on a solution to this without re-joining the same channels multiple times.

My testing shows that the bot joins every public channel in config.json, and errors out when a channel is gone / private without closing the program.

Feedback would be great, thanks!

muggahtee commented:
Looks good @murnux 👍 

From what we have seen, most channels have one of either a public, direct, or ephemeral response, so personally I think the ideal behaviour would be for it to check all three lists, unique them, and then iterate through joining in the way you have implemented

murnux commented:
Thanks for the input! I've pushed the new implementation. In case it isn't clear, here's what I've changed.

- I believe that a different struct for public, dm, eph responses was not necessary because the data types are the exact same. So now they are all under `slackResponse`. It makes using the functions I created easier.
- I added two new functions: `getChannelList` iterates each type of response type, and creates a slice with all of them, including the duplicates. `removeDuplicates` takes the slice created by `getChannelList` and removes duplicates.
- Before the main loop starts, this last slice is iterated and joins channels.

More input is appreciated, I actually went to bed last night thinking: "Wait, I iterated over the responses in the main loop, didn't I?" Haha.

lucymhdavies commented on welcomebot.go:
```diff
@@ -10,28 +10,16 @@ import (
+	PublicResponses []slackResponse `json:"responses"`
+	DmResponses     []slackResponse `json:"dmresponses"`
+	EphResponses    []slackResponse `json:"ephresponses"`
 }
```
Well spotted. Yeah, no need for these to be separate types if they're all the same. :)
